src/storage/seg/src/metrics.rs

```diff
@@ -7,73 +7,172 @@
-    SEGMENT_REQUEST_FAILURE,
-    "number of segment allocation attempts which failed"
-);
-counter!(
-    SEGMENT_REQUEST_SUCCESS,
-    "number of segment allocation attempts which were successful"
-);
-counter!(SEGMENT_EVICT, "number of segments evicted");
-counter!(
-    SEGMENT_EVICT_EX,
-    "number of exceptions while evicting segments"
-);
-counter!(
-    SEGMENT_RETURN,
-    "total number of segments returned to the free pool"
-);
-counter!(SEGMENT_MERGE, "total number of segments merged");
-counter!(SEGMENT_CLEAR, "number of segments cleared");
-counter!(SEGMENT_EXPIRE, "number of segments expired");
-counter!(
-    CLEAR_TIME,
-    "amount of time, in nanoseconds, spent clearing segments"
-);
-counter!(
-    EXPIRE_TIME,
-    "amount of time, in nanoseconds, spent expiring segments"
-);
-gauge!(EVICT_TIME, "time, in nanoseconds, spent evicting segments");
-gauge!(SEGMENT_FREE, "current number of free segments");
-gauge!(SEGMENT_CURRENT, "current number of segments");
+#[metric(
+    name = "segment_request",
+    description = "number of segment allocation attempts"
+)]
+pub static SEGMENT_REQUEST: Counter = Counter::new();
+
+#[metric(
+    name = "segment_request_failure",
+    description = "number of segment allocation attempts which failed"
+)]
+pub static SEGMENT_REQUEST_FAILURE: Counter = Counter::new();
+
+#[metric(
+    name = "segment_request_success",
+    description = "number of segment allocation attempts which were successful"
+)]
+pub static SEGMENT_REQUEST_SUCCESS: Counter = Counter::new();
+
+#[metric(name = "segment_evict", description = "number of segments evicted")]
+pub static SEGMENT_EVICT: Counter = Counter::new();
+
+#[metric(
+    name = "segment_evict_ex",
+    description = "number of exceptions while evicting segments"
+)]
+pub static SEGMENT_EVICT_EX: Counter = Counter::new();
+
+#[metric(
+    name = "segment_return",
+    description = "total number of segments returned to the free pool"
+)]
+pub static SEGMENT_RETURN: Counter = Counter::new();
+
+#[metric(
+    name = "segment_merge",
+    description = "total number of segments merged"
+)]
+pub static SEGMENT_MERGE: Counter = Counter::new();
+
+#[metric(
+    name = "segment_clear",
+    description = "total number of segments cleared"
+)]
+pub static SEGMENT_CLEAR: Counter = Counter::new();
+
+#[metric(
+    name = "segment_expire",
+    description = "total number of segments expired"
+)]
+pub static SEGMENT_EXPIRE: Counter = Counter::new();
+
+#[metric(
+    name = "clear_time",
+    description = "amount of time, in nanoseconds, spent clearing segments"
+)]
+pub static CLEAR_TIME: Counter = Counter::new();
+
+#[metric(
+    name = "expire_time",
+    description = "amount of time, in nanoseconds, spent expiring segments"
+)]
+pub static EXPIRE_TIME: Counter = Counter::new();
+
+#[metric(
+    name = "evict_time",
+    description = "amount of time, in nanoseconds, spent evicting segments"
+)]
+pub static EVICT_TIME: Counter = Counter::new();
```

Comment:
Was the switch to counter intentional? Seems like a reasonable choice since I don't see why it would ever count down, but wanted to check.

Reply:
Yes, I think it was unintentionally a gauge before.